js/form/src/DropdownOptions.svelte

```diff
@@ -10,9 +10,30 @@
 	let distance_from_top: number;
 	let distance_from_bottom: number;
 	let input_height: number;
+	let input_width: number;
 	let refElement: HTMLDivElement;
 	let listElement: HTMLUListElement;
 	let top: string | null, bottom: string | null, max_height: number;
+	let innerHeight: number;
+
+	const calculate_window_distance = () => {
+		distance_from_top = refElement.getBoundingClientRect().top;
+		distance_from_bottom =
+			innerHeight - refElement.getBoundingClientRect().bottom;
```

Comment:
This could be optimized to only get the bounding rect once, like so:

```suggestion
    const {top, bottom} = refElement.getBoundingClientRect(); 
		distance_from_top = innerHeight - top;
		distance_from_bottom = innerHeight - bottom;
```